src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -285,6 +286,14 @@
     @Parameter( defaultValue = "false" )
     private boolean generateUniqueDependencyReducedPom;
 
+    /**
+     * Do we put the dependency reduced pom in the jar instead of the jar file provided by the project.
```

Comment:
This one is odd. If you don't use the DRP you use the JAR?. Shouldn't it read:
Add dependency reduced POM to the JAR instead of the original one provided by the project.

Reply:
That is definitely a typo. Your text is a lot clearer.